code/script.js

```diff
@@ -244,100 +250,151 @@ const generateBoard = () => {
+      value: true, 
+      category: category,
+    }
   }
+  
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// 'Find Out' Function
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const secretValue = secret[currentQuestion.attribute]
+  if (secretValue === currentQuestion.value) {
+    filterCharacters(true)
+  } else {
+    filterCharacters(false) 
+  }
 }
 
-// It'll filter the characters array and redraw the game board.
-const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
+// Filter Section
+const filterCharacters = (keep) => { 
+  const group = currentQuestion.category
+  const attribute = currentQuestion.value 
+ 
   if (group === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`
       )
     }
   } else if (group === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(
+        `Yes, the person is a ${currentQuestion.attribute}! Keep all persons that is a ${currentQuestion.attribute}`
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `No, the person is not a ${currentQuestion.attribute}! Remove all persons that is a ${currentQuestion.attribute}`
+      )
     }
-  }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  } else if (group === 'hair color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${currentQuestion.value} hair! Keep all that has ${currentQuestion.value} hair`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have ${currentQuestion.value} hair! Remove all persons with ${currentQuestion.value} hair`
+      )
+    }
+  } else if (group === 'eye color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${currentQuestion.value} eyes! Keep all that has ${currentQuestion.value} eyes`
+      )
+      
+    } else {
+      alert(
+        `No, the person doesn't have ${currentQuestion.value} eyes! Remove all persons with ${currentQuestion.value} eyes`
+      )
+    }
+  } 
```

Comment:
Great work here getting the filter to work! 💪

There are ways to make the code less repetitive here, i.e. the if, else statements can be written with fewer lines of code. There’s nothing wrong though to start from the basics and write more code - if it makes more sense to you currently - and then move forward slowly towards writing less. But for future reference when you go back and maybe want to touch up on this project for the portfolio, I would recommend you to look at this part again and see if it’s possible to write it with less code 😊